### PR TITLE
E1779:OSS busgandy:Fix teammate advertisement and request to join a team

### PR DESCRIPTION
**Pull request by Suvodeep90**
**Fix Issue #311:** when one respond to an advertisement, (s)he should only be able to respond once. When a team advertises for new members, someone who responds to an ad should only be allowed to respond once. Currently, after someone responds to an ad, they can respond again an unlimited number of times.
**Fixes Done:** When a user responds to an advertisement then a enter is made in the Join team request table with a participant ID which is unique for each user for each assignment. So we are checking if the that particular user has any entry for that assignment with pending or approved status, if so we are not giving the user an option to respond again until the respond to join a team has been declined. 
**Fix Issue #227 :** if user A got a topic and user B got no topic, then A join B’s team, A’s topic be be dropped and A and B end up with a new team with no topic. This issue should be handle carefully because we cannot simply add B to A’s team (imagine, if A has teammate X and B has teammate Y...). One of a potential fix is that, for assignment w/ topics, one cannot not post an ad unless (s)he holds a topic, similarly, one cannot sent invitations unless (s)he holds a topic. 
**Fixes done:** Here we are checking if an assignment has any topic or not, if a assignment don't have any topic then he/she will have option to send out request. If the assignment has topic then we are checking if the user has any topic or not, if user has selected a topic then only he/she will be able to send out a request

### CODE REVIEW

File: app/helpers/student_teams_helper.rb [hunk 1]
```diff
@@ -1,2 +1,16 @@
 module StudentTeamsHelper
+  # This method will return true if the assignment have some topic
+  def topic?(assignment_id)
```
Comment (Winbobob) [inline app/helpers/student_teams_helper.rb]: This method is duplicated to an [existing one](https://github.com/expertiza/expertiza/blob/master/app/models/assignment.rb#L54).

File: app/helpers/student_teams_helper.rb [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
+    topics = SignUpTopic.where("assignment_id = ?", assignment_id)
+    topics.empty?
+  end
+
+  def user_has_topic?(user_id, assignment_id)
```
Comment (Winbobob) [inline app/helpers/student_teams_helper.rb]: It will better to put this method in `team.rb`. Since topics will be assigned to teams.

**Comment by Suvodeep90** [comment]
So do you want me change and push the changes again?

**Comment by ferryxo** [comment]
Yes please. Thanks.

**Comment by Suvodeep90** [comment]
Changed the method location from helper to team.rb

**Comment by Winbobob** [comment]
See #1129

**Pull request closed without merging**